### PR TITLE
fix(lib/associations/has-many.js): COUNT() on HasMany association is …

### LINKED ISSUE #8337: Count on HasMany with associations uses ambiguous name
## What are you doing?
I have a model with a `HasMany` association, and I am trying to count that association.

**NOTE** This issue was fixed in PR #7496, but it was closed as stale without getting merged to master.

```js
const Sequelize = require('sequelize');
const sequelize = new Sequelize('sqlite://:memory:');

const User = sequelize.define('User', {});
const Task = sequelize.define('Task', {});
const TaskItem = sequelize.define('TaskItem', {});

User.hasMany(Task);
Task.belongsTo(User);
Task.hasMany(TaskItem);
TaskItem.belongsTo(Task);


sequelize.sync()
  .then(() => User.create({}))
  .then(user => {
    return user.countTasks({
      include: [
        TaskItem
      ]
    });
  })
  .then(r => console.log(r))
  .catch(err => console.error(err));
```

## What do you expect to happen?
The query successfully executes, returning a count of records associated with the instance.

## What is actually happening?
An exception is thrown when executing the SQL because ``COUNT(`id`)`` is not unique (should use model qualified name, e.g. ``COUNT(`MyModel`.`id`)``). 

```
{ SequelizeDatabaseError: SQLITE_ERROR: ambiguous column name: id
    at Query.formatError (/Users/mike/workspaces/sequelize-hasmany-count-bug/node_modules/sequelize/lib/dialects/sqlite/query.js:418:16)
    at afterExecute (/Users/mike/workspaces/sequelize-hasmany-count-bug/node_modules/sequelize/lib/dialects/sqlite/query.js:119:32)
    at replacement (/Users/mike/workspaces/sequelize-hasmany-count-bug/node_modules/sqlite3/lib/trace.js:19:31)
    at Statement.errBack (/Users/mike/workspaces/sequelize-hasmany-count-bug/node_modules/sqlite3/lib/sqlite3.js:16:21)
  name: 'SequelizeDatabaseError',
  parent: 
   { Error: SQLITE_ERROR: ambiguous column name: id
     errno: 1,
     code: 'SQLITE_ERROR',
     sql: 'SELECT COUNT(`id`) AS `count`, `TaskItems`.`id` AS `TaskItems.id`, `TaskItems`.`createdAt` AS `TaskItems.createdAt`, `TaskItems`.`updatedAt` AS `TaskItems.updatedAt`, `TaskItems`.`TaskId` AS `TaskItems.TaskId` FROM `Tasks` AS `Task` LEFT OUTER JOIN `TaskItems` AS `TaskItems` ON `Task`.`id` = `TaskItems`.`TaskId` WHERE `Task`.`UserId` = 1;' },
  original: 
   { Error: SQLITE_ERROR: ambiguous column name: id
     errno: 1,
     code: 'SQLITE_ERROR',
     sql: 'SELECT COUNT(`id`) AS `count`, `TaskItems`.`id` AS `TaskItems.id`, `TaskItems`.`createdAt` AS `TaskItems.createdAt`, `TaskItems`.`updatedAt` AS `TaskItems.updatedAt`, `TaskItems`.`TaskId` AS `TaskItems.TaskId` FROM `Tasks` AS `Task` LEFT OUTER JOIN `TaskItems` AS `TaskItems` ON `Task`.`id` = `TaskItems`.`TaskId` WHERE `Task`.`UserId` = 1;' },
  sql: 'SELECT COUNT(`id`) AS `count`, `TaskItems`.`id` AS `TaskItems.id`, `TaskItems`.`createdAt` AS `TaskItems.createdAt`, `TaskItems`.`updatedAt` AS `TaskItems.updatedAt`, `TaskItems`.`TaskId` AS `TaskItems.TaskId` FROM `Tasks` AS `Task` LEFT OUTER JOIN `TaskItems` AS `TaskItems` ON `Task`.`id` = `TaskItems`.`TaskId` WHERE `Task`.`UserId` = 1;' }

```

__Dialect:__ sqlite 
__Dialect version:__ 3.1.11
__Database version:__ 
__Sequelize version:__ 4.8.4
__Tested with master branch:__ Yes

**Note :** _Your issue may be ignored by maintainers if its not tested against `master` branch OR does not follow issue template._

### PR DESCRIPTION
**Pull request by yachaka**
…not ambiguous anymore

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)? **No API change**
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](../CONTRIBUTING.md)?

### Description of change

HasMany.count() was counting on the target model primaryKeyField without including the target table
name. This led to ambiguous column error when both Models (the source and the target) had this field
defined.

fix #8337

---

Also, on a side note, there is already some PRs that were opened to try and solve this issue (one already by me):

#7496 // April 6 2017
#6807 // Nov 4 2016
#6115 // June 15 2016

And some issues related to this bug:

#8489 // Oct 16 2017
#8337 // Sept 19 2017
#6488 // Aug 23 2016

Let's get this merged!

### CODE REVIEW

File: test/integration/associations/has-many.test.js [hunk 1]
```diff
@@ -26,6 +26,35 @@ describe(Support.getTestDialectTeaser('HasMany'), () => {
     });
   });
 
+  describe('count', () => {
+    it('the COUNT() attribute should be `{Model.name}.{Model.primaryKeyField}`', function() {
```
Comment (sushantdhiman) [inline test/integration/associations/has-many.test.js]: No stubs, its should be a proper integration test hitting database and asserting on returned data

Comment (sushantdhiman) [reply test/integration/associations/has-many.test.js]: Or use this [year old test](https://github.com/sequelize/sequelize/pull/6115#discussion_r69303807) as suggested by @janmeier

**Comment by yachaka** [comment]
@sushantdhiman Used the test written by @janmeier.
Please review again 👍

**Comment by sushantdhiman** [comment]
Need additional review from @janmeier due to his involvement in last discussion for these PRs

File: test/integration/associations/has-many.test.js [hunk 2]
```diff
@@ -26,6 +26,44 @@ describe(Support.getTestDialectTeaser('HasMany'), () => {
...
+        return expect(user.countTasks({
+          attributes: [Task.primaryKeyField, 'title'],
+          include: [{
+            attributes: [],
+            association: subtasks,
+            where: {
+              active: true
+            }
+          }],
+          group: this.sequelize.col(Task.name.concat('.', Task.primaryKeyField))
```
Comment (janmeier) [inline test/integration/associations/has-many.test.js]: `group: [Task.primaryKeyField]` should work as well

**Pull request merged**